OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,310 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Adds a particle-based overlay.")]
+	public class WeatherOverlayInfo : ITraitInfo
+	{
+		[Desc("Factor for particle density. As higher as more particles will get spawned.")]
+		public readonly float ParticleDensityFactor = 0.0007625f;
+
+		[Desc("Should the level of the wind change over time, or just stick to the first value of WindLevels?")]
+		public readonly bool ChangingWindLevel = true;
+
+		[Desc("The levels of wind intensity (particles x-axis movement in px/tick).")]
+		public readonly int[] WindLevels = { -5, -3, -2, 0, 2, 3, 5 };
+
+		[Desc("Works only if ChangingWindLevel is enabled. Min. and max. ticks needed to change the WindLevel.")]
+		public readonly int[] WindTick = { 150, 750 };
+
+		[Desc("Hard or soft fading between the WindLevels.")]
+		public readonly bool InstantWindChanges = false;
+
+		[Desc("Particles are drawn in squares when enabled, else with lines.")]
+		public readonly bool UseSquares = true;
+
+		[Desc("Works only with squares enabled. Size min. and max. value in pixels.")]
+		public readonly int[] PaticleSize = { 1, 3 };
+
+		[Desc("Scatters falling direction on the x-axis. Scatter min. and max. value in px/tick.")]
+		public readonly int[] ScatterDirection = { -1, 1 };
+
+		[Desc("Min. and max. speed at which particles fall in px/tick.")]
+		public readonly float[] Gravity = { 1.00f, 2.00f };
+
+		[Desc("The current offset value for the swing movement. SwingOffset min. and max. value in px/tick.")]
+		public readonly float[] SwingOffset = { 1.0f, 1.5f };
+
+		[Desc("The value that particles swing to the side each update. SwingSpeed min. and max. value in px/tick.")]
+		public readonly float[] SwingSpeed = { 0.001f, 0.025f };
+
+		[Desc("The value range that can be swung to the left or right. SwingAmplitude min. and max. value in px/tick.")]
+		public readonly float[] SwingAmplitude = { 1.0f, 1.5f };
+
+		[Desc("The randomly selected ArgbColors for the particles. Use this order: a,r,g,b,  a,r,g,b, a,...")]
```

Comment:
This isn't quite what I meant. `ARGB` already tells the order.
Modders don't know what the colors are for. Moreso why do you want an array rather than a single value?